src/Monolog/Logger.php

```diff
@@ -356,8 +365,9 @@ public function addRecord(int|Level $level, string $message, array $context = []
         try {
             $recordInitialized = \count($this->processors) === 0;
 
+            $datetime = $useClock ? $this->clock->now() : new JsonSerializableDateTimeImmutable($this->microsecondTimestamps, $this->timezone);
```

Comment:
phpstan is right about the issue reported here: if the caller passes `true` for `$useClock`, this will break when the clock is `null`.

And this is also quite useless when using the PSR-3 interface as those would never pass `true` there.